.github/workflows/unit_tests.yml

```diff
@@ -35,11 +35,7 @@ jobs:
           python -m pip install --upgrade pip
           pip install flake8 pytest pytest-cov
           pip install psutil joblib
-          if [ "${{ matrix.os }}" == "ubuntu-latest" ]; then
-            pip install mpi4py
-          else
-            conda install --yes mpi4py openmpi
-          fi
+          conda install --yes -c conda-forge mpi4py openmpi
```

Comment:
Note that this was changed because the mpi4py installation was failing for Ubuntu/Python 3.9. Not totally sure what the root of the issue was.

Reply:
Assuming the default repository identified for the conda install had an old/incompatible version?

Reply:
Something like that. We technically shouldn't need to install the openmpi python library on Linux if it's already installed at the system level, but now we can just rely on wheels that are compiled and tested by the maintainers of mpi4py.